src/Pact/Native.hs

```diff
@@ -153,6 +156,19 @@ enforceDef = defNative "enforce" enforce
     enforce' i as = argsError i as
     {-# INLINE enforce' #-}
 
+    enforceLazy i [cond, msg] = do
+      cond' <- reduce cond
+      case cond' of
+        TLiteral (LBool b') _ ->
+          if b' then
+            return (TLiteral (LBool True) def)
+          else reduce msg >>= \case
```

Comment:
I like this idiom, so to be consistent we should use it above too: `reduce cond >>= \case ...`.

Reply:
Really good catch!! Yes I agree.